packages/cli/src/commands/generate/__tests__/helpers.test.js

```diff
@@ -433,3 +433,57 @@ test('ensureUniquePlural skips any rule if singular and plural are already diffe
   expect(pluralize.singular(singular)).toBe(singular)
   expect(pluralize.plural(singular)).toBe(plural)
 })
+
+// mapRouteParamTypeToTsType tests
+test('mapRouteParamTypeToTsType maps scalar type String to TS type string', () => {
+  expect(helpers.mapRouteParamTypeToTsType('String')).toBe('string')
+})
+
+test('mapRouteParamTypeToTsType maps scalar type Boolean to TS type boolean', () => {
+  expect(helpers.mapRouteParamTypeToTsType('Boolean')).toBe('boolean')
+})
+
+test('mapRouteParamTypeToTsType maps scalar type Int to TS type number', () => {
+  expect(helpers.mapRouteParamTypeToTsType('Int')).toBe('number')
+})
+
+test('mapRouteParamTypeToTsType maps scalar type Float to TS type number', () => {
+  expect(helpers.mapRouteParamTypeToTsType('Float')).toBe('number')
+})
+
+test('mapRouteParamTypeToTsType maps unexpected type to TS unknown', () => {
+  expect(helpers.mapRouteParamTypeToTsType('unknown')).toBe('unknown')
+})
+
+// mapPrismaScalarTypeToTsType tests
+test('mapPrismaScalarTypeToTsType maps scalar type String to TS type string', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('String')).toBe('string')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type Boolean to TS type boolean', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('Boolean')).toBe('boolean')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type Int to TS type number', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('Int')).toBe('number')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type BigInt to TS type number', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('BigInt')).toBe('number')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type Float to TS type number', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('Float')).toBe('number')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type Decimal to TS type number', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('Float')).toBe('number')
+})
+
+test('mapPrismaScalarTypeToTsType maps scalar type DateTime to TS type Date', () => {
+  expect(helpers.mapPrismaScalarTypeToTsType('DateTime')).toBe('Date')
```

Comment:
A few of things here. 

1. I think the function should be named more specific `mapPrismaScalarToPagePropType` or something similar. Because its not meant to be a general converter, but just for page props.
2. I don't think DateTime -> Date because we don't parse Dates as part of the router (atleast for now)
3. (not important) Would it be possible to group these tests in a describe block please? Would be a little easier to read e.g.
```js
describe('mapPrismaxxx', () => {
it ('maps Float to number') {
}
it ('maps String to string') {
}
})
```

Reply:
1. Right, renamed to `mapPrismaScalarToPagePropTsType`.
2. DateTime -> string now.
3. Done!